File: day3_part1_2016.f90
program day3
  implicit none
  integer :: a, b, c, count
  open(unit=10, file='input.txt', status='old')
  count = 0
  do while(.true.)
    read(10, *, iostat=a) a, b, c
    if (a < 0) exit
    if (a + b > c .and. a + c > b .and. b + c > a) then
      count = count + 1
    end if
  end do
  close(10)
  print*, count
end program day3

File: pascal/day3_part1_2016.pas
program Day3;

var
    inputFile: Text;
    a, b, c, countPossibleTriangles: Integer;

begin
    assign(inputFile, 'input.txt');
    reset(inputFile);

    countPossibleTriangles := 0;

    while not eof(inputFile) do
    begin
        readln(inputFile, a, b, c);

        if (a + b > c) and (a + c > b) and (b + c > a) then
            countPossibleTriangles := countPossibleTriangles + 1;
    end;

    close(inputFile);

    writeln(countPossibleTriangles);
end.